.semaphore/semaphore.yml

```diff
@@ -0,0 +1,88 @@
+---
+version: "v1.0"
+name: "Vaultenv CI Pipeline"
+
+agent:
+  machine:
+    type: "e1-standard-2"
+    os_image: "ubuntu2004"
+
+blocks:
+  - name: "Run checks"
+    task:
+      prologue:
+        commands:
+          - "checkout"
+
+          # Haskell requires libgmp for big integers.
+          - "sudo apt install libgmp-dev"
+
+          # TODO: verify installation script
+          # Install Stack from source
+          - "mkdir -p ~/.local/bin"
+          - "export PATH=$HOME/.local/bin:$PATH"
+          - "wget https://github.com/commercialhaskell/stack/releases/download/v2.7.3/stack-2.7.3-linux-x86_64.tar.gz"
+          - "sha256sum --check .semaphore/install-stack.2.7.3.sha256"
+          - "cat stack-2.7.3-linux-x86_64.tar.gz | tar xz --wildcards --strip-components=1 -C ~/.local/bin '*/stack'"
+
+          # restore stack cache
+          - "cache restore stack-"
+
+      jobs:
+        - name: "Run tests for client"
+          commands:
+            - "(cd client-haskell && stack setup -j2 --no-terminal)"
+            - "(cd client-haskell && stack build -j2 --no-terminal)"
+            - "(cd client-haskell && stack test  -j2 --no-terminal)"
+
+        - name: "Run tests for server"
+          commands:
+            - "(cd server && stack setup -j2 --no-terminal)"
+            - "(cd server && stack build -j2 --no-terminal)"
+            - "(cd server && stack test  -j2 --no-terminal)"
+
+      epilogue:
+        always:
+          commands:
+            # Store a copy of the stack installation folder
+            - "cache store stack-$(date -u -Idate) ~/.stack"
+
+  - name: "Build"
+    task:
+      prologue:
+        commands:
+          - "checkout"
+
+          # Download and verify Nix installation script.
+          - curl -o install-nix-2.3.15 https://releases.nixos.org/nix/nix-2.3.15/install
+          - sha256sum --check .semaphore/install-nix-2.3.15.sha256
+
+          # Restore `/nix` cache. Create the directory first, otherwise we encounter
+          # permission errors. We do this because the Semaphore cache is faster than
+          # both Cachix and cache.nixos.org.
+          - "sudo mkdir /nix"
+          - "sudo chown -R semaphore:semaphore /nix"
+          - "cache restore nix-store-"
```

Comment:
By restoring *any* nix-store cache we have, the nix store will only ever grow until we run out of disk at some point. Since we're not running a lot of builds on this repo, this is probably fine, but something to keep in mind. In our other nix-based projects we only restore the nix-store cache from the same day, so the first build on each day takes a bit longer.

Reply:
You have a point, but for this repo I'll leave it as is